Composed listenables fails in Opera

According to [testling](https://ci.testling.com/spoike/reflux), the following test fails in Opera and older versions of Chrome:

```
not ok 8 Composed listenables should emit multiple times
  Error: Uncaught AssertionError: expected [] to deeply equal [ Array(2) ] (http://git.testling.com/work/spoike/repos/ba944a00a042aeeb2ec80d119d93684beb9ff196.1407304513039/1407304513279.c3212e4f.js:2319)
      at global.onerror (http://git.testling.com/work/spoike/repos/ba944a00a042aeeb2ec80d119d93684beb9ff196.1407304513039/node_modules/mocha/mocha.js:5871:10)
```

--- a/test/composedListenable.spec.js
+++ b/test/composedListenable.spec.js
@@ -69,11 +69,16 @@ describe('Composed listenables', function() {
     });
 
 
-    it('should emit multiple times', function(done) {
-        var callArgs = [];
-        all.listen(function() {
-            callArgs.push([].slice.call(arguments));
-        }, null);
+    it('should emit multiple times', function() {
+        var promise = Q.promise(function(resolve) {
+            var callArgs = [];
+            all.listen(function() {
+                callArgs.push([].slice.call(arguments));
+                if (callArgs.length === 2) {
+                    resolve(callArgs);
+                }
+            });
+        });
 
         action1('a');
         action2('b');
@@ -83,13 +88,10 @@ describe('Composed listenables', function() {
         action2('y');
         action3('z');
 
-        setTimeout(function() {
-            assert.deepEqual(callArgs, [
-                [['a'], ['b'], ['c']],
-                [['x'], ['y'], ['z']]
-            ]);
-            done();
-        }, 200);
+        return assert.eventually.deepEqual(promise, [
+          [['a'], ['b'], ['c']],
+          [['x'], ['y'], ['z']]
+        ]);
     });
 
 

